internal/db/schema/migrations/oss/postgres/80/11_host_catalog_base_table_updates.up.sql

```diff
@@ -0,0 +1,59 @@
+-- Copyright (c) HashiCorp, Inc.
+-- SPDX-License-Identifier: BUSL-1.1
+
+begin;
+
+  -- Add create_time and update_time to host_catalog table.
+  alter table host_catalog add column create_time wt_timestamp;
+  alter table host_catalog add column update_time wt_timestamp;
+
+  -- Update rows with current values
+  update host_catalog
+    set create_time = sgc.create_time, update_time = sgc.update_time
+    from host_catalog as hc
+    left join static_host_catalog as sgc on hc.public_id = sgc.public_id;
+  update host_catalog
+    set create_time = hpc.create_time, update_time = hpc.update_time
+    from host_catalog as hc
+    left join host_plugin_catalog as hpc on hc.public_id = hpc.public_id;
+
+  -- Replace the insert trigger to also set the create_time
+  -- Replaces the insert_host_catalog_subtype function defined in 44/02_hosts.up.sql
+   create or replace function insert_host_catalog_subtype() returns trigger
+  as $$
+  begin
+    insert into host_catalog
+      (public_id, project_id, name, create_time)
+    values
+      (new.public_id, new.project_id, new.name, new.create_time);
+    return new;
+  end;
+  $$ language plpgsql;
+
+  -- Add trigger to update the new update_time column on every subtype update.
+  create function update_host_catalog_table_update_time() returns trigger
+  as $$
+  begin
+    update host_catalog set update_time = now() where public_id = new.public_id;
```

Comment:
I'm a little wary to change this since it depends on the trigger that sets the update time automatically on the subtype table to run before this trigger does, and I'm not sure what the order of the triggers are or if we can rely on it to be consistent over time. This feels like the safer option and, as you say, it should be equivalent.